### DISCUSSION TITLE
Returning `Outcome::Error` from `from_request`

### DISCUSSION THREAD

**Discussion by jcuenod** [rwf2/Rocket#2688] [Questions]
I'm implementing `FromRequest` on my `SearchParams` struct, but I'm having trouble with the `from_request` parser I'm writing.

I want to return some kind of error json with an appropriate status code and a message. Something like:

```
return Outcome::Error((
    rocket::http::Status::BadRequest,
    rocket::Error::from("Invalid page size"),
));
```

I can't figure out how to work with `Outcome::Error` and I keep finding older documentation about `Outcome::Failure` that looks closer to what I'm wanting to do. How do I do this?

**Comment by SergioBenitez** [comment]
> I keep finding older documentation about Outcome::Failure that looks closer to what I'm wanting to do. How do I do this?

`Failure` was renamed to `Error`. The name was the only thing that changed. Your question doesn't really include what you're trying to do, just what you tried. I suspect what you're trying to do isn't actually going to help you
accomplish what you want. Based on your question, my best advice would be to read through the official guide, if you haven't already, or give it a second look if you have. 

> I want to return some kind of error json with an appropriate status code and a message. Something like

Request guards cannot respond directly to a request. They can signal an error outcome, and a handler can catch that by using `Result<T, E>` where `T` is the type of the request guard and `E` is the type of the error value in the `Outcome::Error`. The handler can then return the error as a response if it likes. 

> return Outcome::Error((
    rocket::http::Status::BadRequest,
    rocket::Error::from("Invalid page size"),
));

It's very odd to use Rocket's error type as the error value in your request guard. No documentation should have led you to that. You should let the types guide you. Or the examples. Please read our documentation here: https://api.rocket.rs/v0.5/rocket/request/trait.FromRequest.html#example-1

**Comment by jcuenod** [comment]
Thanks, I hadn't realized that I could use `Result` in my function definition and figure out the error that way. You are right, I had no idea how to implement the `Outcome::Error`, so that's why I was using Rocket's error type. Thanks for the help!
